docs/wiki/Getting Started With Fuzzing.md

```diff
@@ -0,0 +1,159 @@
+fuzz: elapsed: 0s, gathering baseline coverage: 7/7 completed, now fuzzing with 8 workers
+fuzz: elapsed: 1s, execs: 10 (18/sec), new interesting: 2 (total: 9)
+--- PASS: FuzzDecrypt (0.57s)
+PASS
+ok      decred.org/dcrdex/dex/encrypt   1.172s
+```
+
+## Failing Input
+
+Failing inputs generated by the fuzzing engine are saved to the main directory of the project (e.g `dcrdex/testData/fuzz/$fuzzTarget`), and are re-run as part of `go test` (without `-fuzz FuzzTarget`) for `$fuzzTarget`.
+
+A failure may occur while fuzzing for several reasons:
+
+- A panic occurred in the code or the test.
+- The fuzz target called t.Fail, either directly or through methods such as t.Error or t.Fatal.
+- A non-recoverable error occurred, such as an os.Exit or stack overflow.
+- The fuzz target took too long to complete. Currently, the `timeout for an execution of a fuzz target is 1 second`. This may fail due to a deadlock or infinite loop, or from intended behaviour in the code.
+
+## Other Highlighted Fuzz Flags
+
+- **-fuzztime**:
+    The total time or number of iterations that the fuzz target will be executed before exiting, specified as a time.Duration (for example, `-fuzztime 1h30s`) or using aspecial syntax Nx to run the fuzz target N times (for example, `-fuzztime 1000x`). The default is to run forever.
+- **-fuzzminimizetime**: the time or number of iterations that the fuzz target will be executed during each minimization attempt, default 60sec. You can completely disable minimization by setting -fuzzminimizetime 0 when fuzzing.
+- **-parallel**: the number of fuzzing processes running at once, default $GOMAXPROCS. Currently, setting -cpu during fuzzing has no effect.
+- **-keepfuzzing**: Keep running the fuzz test if a crasher is found. (default false)
+
+## Best Practices
+
+- Fuzz targets `should be fast and deterministic` so the fuzzing engine can work efficiently, and new failures and code coverage can be easily reproduced.
+ Fuzzing is most effective with the most primitive methods and functions. As such, this will take some careful targeting and potentially some very light refactoring here and there to get fuzzable functions.
+- Since the fuzz target is invoked in parallel across multiple workers and in nondeterministic order, the state of a fuzz target should not persist past the end of each call, and the behaviour of a fuzz target `should not depend on a global state`.
+- Keeping auto-generated seed corpus in $GOCACHE increases the efficiency of the fuzzing engine. The more seed corpus the better.
+- Provide sufficient seed corpus. This could be in the test file, another file or a directory. This will be a guide to the fuzzing engine in generating sensible random inputs.
+To convert provided seed corpus files to Go fuzzing corpus format use [file2fuzz](https://pkg.go.dev/golang.org/x/tools/cmd/file2fuzz):
+
+  ```sh
+
+	$ go install golang.org/x/tools/cmd/file2fuzz@latest
+	$ file2fuzz
+
+  ```
```

Comment:
What do I do with file2fuzz?

Reply:
```text
To convert provided seed corpus files to Go fuzzing corpus format use [file2fuzz]
```

Reply:
TBH this still isn't clear to me.  I can see that file2fuzz takes some binary file as input and outputs to the "Go fuzzing corpus format".  But what kind of input fie?  Then where do you put the output and how do you use it?

Reply:
`file2fuzz` sounds like an alternative to using a random string of your choice to go in the `[]byte`.  Basically it puts the contents of the file into a corpus like:

```
go test fuzz v1
[]byte("<contents of file>")
```

And then you... ? put that somewhere for the go fuzz test to find and use?

Reply:
> And then you... ? put that somewhere for the go fuzz test to find and use?

Yes, `file2fuzz` has an `-o` flag that provides a path to write output(seed corpus in Go fuzzing corpus format) files to.